lib/inc/internal/config_impl_util.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace hocon {
+
+    class config_impl_util {
+    public:
+        static bool equals_handling_null(std::string const& a, std::string const& b);
```

Comment:
Question here: the original Java implementation of these methods is generic:

```
    static boolean equalsHandlingNull(Object a, Object b) {
        if (a == null && b != null)
            return false;
        else if (a != null && b == null)
            return false;
        else if (a == b) // catches null == null plus optimizes identity case
            return true;
        else
            return a.equals(b);
    }
```

From what I've seen so far, it looks like we're only going to do this with strings and vectors, so I opted to just overload it for those types. Is this okay, or do we need to take a more generic approach?